### PR TITLE
Make Elm highlighting the default (for code blocks without language tag)

### PR DESCRIPTION
**Pull request by jvoigtlaender**
Replaces https://github.com/evancz/elm-markdown/pull/7.

Merging this will have the following consequences.

For users of  `elm-markdown`:
- If they aren't binding `highlight.js` in their app, no consequences at all.
- If they are binding `highlight.js`, but are only ever producing code blocks with explicit language tags, no consequences at all.
- If they are binding `highlight.js`, and are producing code blocks without language tags, but have not included "elm" in the set of available languages chosen for the specific version of `highlight.js` bound (presumably because their app has no desire to highlight any Elm code), no consequences at all.
- If they are binding `highlight.js`, and the bound version of that library has "elm" in its set of available languages, and they are producing code blocks without language tags, then while previously these code blocks would have received no syntax highlighting, they will now be syntax highlighted as Elm code.

So, very minor consequences, which moreover can be easily evaded/opted-out of (by using a non-existing language tag like `xxx` for code that should not be highlighted, or by not including "elm" in highlight.js if it is not wanted). Plus, until very recently `elm-markdown`'s use of `highlight.js` was completely undocumented, so there should be no code relying on this out there at the moment (except for the package.elm-lang.org repo).

For http://package.elm-lang.org/:
- If recompiled with the changed version of `elm-markdown`, all untagged code blocks on the documentation pages, in particular all inline code, will receive the desired Elm syntax highlighting. See https://github.com/evancz/elm-markdown/pull/7#issuecomment-124731397 for an example of how this will look.

In summary, this pull request should achieve the ultimate goal of https://github.com/elm-lang/package.elm-lang.org/issues/39, without bad consequences for other use(r)s of `elm-markdown`.

### CODE REVIEW

**Comment by evancz** [comment]
I think someone making a ruby or python website should be able to have whatever default they want. What about adding a "defualt language" to the [`Options`](http://package.elm-lang.org/packages/evancz/elm-markdown/1.1.5/Markdown#Options) type so people can decide for themselves?

**Comment by jvoigtlaender** [comment]
I suggested having something like that in https://github.com/evancz/elm-markdown/issues/4. But I have no idea how to do it myself, since the way these options are passed to the JS library seems not completely straightforward.

**Comment by jvoigtlaender** [comment]
Actually, that specific suggestion is in [this mailing list message](https://groups.google.com/d/msg/elm-discuss/lBXRpf3BQiw/Z9YIxn5UPnAJ), not in that PR. Anyway, seems a good idea to me.

**Comment by evancz** [comment]
Yeah, as soon as we make a default, I bet there will be issues open complaining about the default ;)

I'll bring this up at the elm meetup tonight and see if someone maybe wants to take a look. Though I maybe should just do it, that might be faster.

**Comment by jvoigtlaender** [comment]
Since it doesn't look like there is bandwidth for implementing a customization option for setting the default language here at the moment, can I suggest to simply go with making Elm the default?

Reasoning:
- It's such a simple change (all here in this PR).
- It would be so beneficial, since immediately (after recompiling https://github.com/elm-lang/package.elm-lang.org once with the new version of `elm-markdown` resulting here) we would get source code highlighting for all code examples in all package documentations on http://package.elm-lang.org/.
- It's not hurting anyone at the moment. If those people potentially making a ruby or python website show up, they will be quick enough to demand an option to change the default-highlighted language. At that point, there will be enough motivation to actually implement that customization mechanism here. At the moment there clearly isn't. But should that keep us from getting the above benefits indefinitely?

**Comment by evancz** [comment]
I just hacked the package.elm-lang.org website to have this behavior. I don't really want to do this to the main package though. Now that package.elm-lang.org is defaulting to Elm, do you still think this is a good idea?

**Comment by jvoigtlaender** [comment]
Depends on how hacky the hack to package.elm-lang.org is. :smile:

Will take a look.

**Comment by jvoigtlaender** [comment]
Ah, I don't see your hack in the https://github.com/elm-lang/package.elm-lang.org repo. So you did it only in the live version? Will it survive the next update of one of the markdown dependencies (the https://github.com/chjj/marked library, the https://highlightjs.org/ library, `elm-markdown`)?

**Comment by evancz** [comment]
I essentially merged in your patch, just for the website though. In that I consider the proposal here a hack, this method means that it does not effect the public API and behavior of elm-markdown but solves the specific problem we care about. I think this is a better balance. If you don't like this solution, I think the other route is the right answer and it's not gonna be too hard.

**Comment by jvoigtlaender** [comment]
Ah, so you did what I did locally to test my patch to `elm-markdown`. (I copied the `elm-markdown` sources into my local `package.elm-lang.org` repo, made the change in source, then compiled.)

That's okay. But won't you regret that next time `elm-markdown` changes for some other reason and you then want to use that new version for `package.elm-lang.org`?

Ultimately, at the moment I care about having syntax highlighting on the packages site. Since I care about everything that makes Elm more friendly to newcomers. If you can maintain the "hack" for that, that's fine with me, and I'll happily close this issue against `elm-markdown` here. As for the "right way" to do it in the long term, I agree that proper customization would be better for that. But this brings us back to where we were two months ago. Back then, I had looked at doing it myself, but failed to see how, since the way the options are passed is doing something evil imperative (as in "imperative programming").

**Comment by evancz** [comment]
Yeah, it'd be lame if it broke again, so let's do it the right way.

I believe you can set the options all at once instead of doing it partially like I did. You just gotta read [this](https://github.com/chjj/marked#usage) a bit. So [the options](https://github.com/chjj/marked#options-1) are where you set highlighting, and [the marked function](https://github.com/chjj/marked#markedmarkdownstring-options-callback) can take that as an argument. So you just give the options when you parse instead of setting them globally.

It's true that reading JS APIs is silly and mysterious, but I don't think it's so crazy to do the right thing here.

**Comment by jvoigtlaender** [comment]
Okay, I'll look into this. Maybe I can get over what worried me last time. :smile:

**Comment by evancz** [comment]
Cool :) Let me know if you run into anything weird. I mainly think it requires getting rid of the `setOptions` thing and adding a `highlight` field when creating the options object. Not 100% sure though.

**Comment by evancz** [comment]
Let's do a new PR for this as well.

**Comment by jvoigtlaender** [comment]
The new pull request is https://github.com/evancz/elm-markdown/pull/10.
